src/common/report.interfaces.ts

```diff
@@ -9,8 +9,8 @@ export interface Report {
 
 export interface FileReport extends Report {
   kind: 'file';
-  errors: SyntaxError[];
-  warnings: unknown[];
+  errors: (SyntaxError | ValidationIssue)[];
+  warnings: ValidationIssue[];
```

Comment:
Don't we want to create new type of `Report` interface specifically for examples validation?

Reply:
I don't think so, we would need to further modify the format functions .